src/models/queues.ts

```diff
@@ -447,5 +468,21 @@ QueueSchema.method<QueueDocument>("getWaitingRooms", function (guild: GuildDocum
     return guild.voice_channels?.filter(x => x.queue?.equals(this._id!)) ?? [];
 });
 
+QueueSchema.method<QueueDocument>("subscribeOnJoinWhenEmpty", function (discord_id: string, callback_function:() => void): boolean {
+    if (this.callbackFunctions.has(discord_id)){
+        return false;
+    }
+    this.callbackFunctions.set(discord_id, callback_function);
```

Comment:
`.set()` markiert einen Wert nur zum speichern vor. Um in die DB zu schreiben fehlt hier noch:
```ts
await this.$parent()?.save();
```
(und eine function kann nicht direkt in der DB gespeichert werden)